src/replication.c

```diff
@@ -418,13 +418,13 @@ void feedReplicationBuffer(char *s, size_t len) {
         }
         if (add_new_block) {
             createReplicationBacklogIndex(listLast(server.repl_buffer_blocks));
+
+            /* It is important to trim after adding replication data to keep the backlog size close to
+            * repl_backlog_size in the common case. We wait until we add a new block to avoid repeated
+            * unnecessary trimming attempts when small amounts of data are added. See comments in
+            * freeMemoryGetNotCountedMemory() for details on replication backlog memory tracking. */
```

Comment:
```suggestion
             * repl_backlog_size in the common case. We wait until we add a new block to avoid repeated
             * unnecessary trimming attempts when small amounts of data are added. See comments in
             * freeMemoryGetNotCountedMemory() for details on replication backlog memory tracking. */
```